pages/solana.tsx

```diff
@@ -0,0 +1,347 @@
+import DoneFilledIcon from "../components/UI/iconsComponents/icons/doneFilledIcon";
+import theme from "../styles/theme";
+import DomainActions from "../components/solana/domainActions";
+import DiscountEndScreen from "../components/discount/discountEndScreen";
+import identityChangeCalls from "../utils/callData/identityChangeCalls";
+
+const Solana: NextPage = () => {
+  const { address: starknetAddress } = useAccount();
+  const { publicKey: solPublicKey, signMessage } = useWallet();
+  const { addTransaction } = useNotificationManager();
+  const [walletModalOpen, setWalletModalOpen] = useState<boolean>(false);
+  const [currentStep, setCurrentStep] = useState<number>(0);
+  const [snsDomains, setSnsDomains] = useState<string[]>([]);
+  const [selectedDomain, setSelectedDomain] = useState<SnsDomainData>();
+  const [hasLoadedSolDomain, setHasLoadedSolDomain] = useState<boolean>(false);
+  const [isTxModalOpen, setIsTxModalOpen] = useState(false);
+  const claimedDomains = useHasClaimSolSubdomain(
+    snsDomains,
+    starknetAddress as string
+  );
+  const [callData, setCallData] = useState<Call[]>([]);
+  const { writeAsync: execute, data: registerData } = useContractWrite({
+    calls: callData,
+  });
+  const [open, setOpen] = useState(true);
+
+  useEffect(() => {
+    const currentDate = new Date();
+    const timestamp = currentDate.getTime();
+
+    if (timestamp >= 1706878847 * 1000) {
+      setOpen(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!starknetAddress) setCurrentStep(0);
+    else if (starknetAddress && !solPublicKey) setCurrentStep(1);
+    else if (starknetAddress && solPublicKey) setCurrentStep(2);
+  }, [starknetAddress, solPublicKey]);
+
+  // fetch sol domains from SNS api
+  useEffect(() => {
+    if (!solPublicKey) return;
+    fetch(`https://sns-api.bonfida.com/owners/${solPublicKey}/domains`)
+      .then((response) => response.json())
+      .then((res) => {
+        if (res?.success && res.result) setSnsDomains(res.result);
+        else setSnsDomains([]);
+        setHasLoadedSolDomain(true);
+      })
+      .catch((error) => {
+        console.log("An error occured while fetching user SNS domains", error);
+        setSnsDomains([]);
+        setHasLoadedSolDomain(true);
+      });
+  }, [solPublicKey]);
+
+  useEffect(() => {
+    if (!selectedDomain || !selectedDomain.signature) return;
+    const name = utils.encodeDomain(selectedDomain.name);
+    setCallData([
+      SolanaCalls.claimDomain(
+        name[0].toString(),
+        selectedDomain.signature.r,
+        selectedDomain.signature.s,
+        selectedDomain.signature.max_validity as number
+      ),
+      SolanaCalls.setResolving(name[0].toString(), starknetAddress as string),
+      SolanaCalls.setAsMainDomain(name[0].toString()),
+    ]);
+  }, [selectedDomain, starknetAddress]);
+
+  useEffect(() => {
+    if (!registerData?.transaction_hash) return;
+    // posthog?.capture("register");
```

Comment:
should we add this event? might be useful no?

Reply:
@fricoben which event should be added when a user claims a sol domain ?

Reply:
No